### ISSUE TITLE
issues with sanity_check_paths

### ISSUE THREAD

**Issue by andreas-h** [hpcugent/easybuild-framework#2196]
I'm having an issue with `sanity_check_paths`:

When I'm setting 

    sanity_check_paths = {
        'files': ['bin/emos_tool', 'lib/libemos.a'],
    }

I get the error *Incorrect format for sanity_check_paths (should (only) have dirs,files keys, values should be lists (at least one non-empty)).*

However, when I set

    sanity_check_paths = {
        'files': ['bin/emos_tool', 'lib/libemos.a'],
        'dirs': [],
    }

everything works fine.

I cannot judge if the problem is the error message or the sanity_check mechanism (in the CMakeMake easyblock), but the current behaviour seems wrong to me.

**Comment by andreas-h** [comment]
A second point is that this error is only raised *after* installing.  This is not optimal, as it wastes (potentially) a lot of time.  Checks like this should be performed before even configuring.

**Comment by boegel** [comment]
@andreas-h The requirement of specifying both `files` and `dirs` is intentional, to keep things consistent.

You're right that the error should be raised earlier, we already have an open issue for that, cfr. #703.

Are you up for taking a look into fixing this?

**Comment by andreas-h** [comment]
Yes, can do that.  It'll be some time before I can do so, though.
